src/OSPSuite.Infrastructure.Import/Services/Importer.cs

```diff
@@ -173,16 +178,25 @@ public MappingProblem CheckWhetherAllDataColumnsAreMapped(ColumnInfoCache dataCo
 
          return new MappingProblem()
          {
+            //all the mandatory mappings that have not been mapped to a column
             MissingMapping = dataColumns
                .Where(col => col.IsMandatory && subset.All(cm =>
                   cm.MappedColumn.Name != col.Name)).Select(col => col.Name)
                .ToList(),
-            MissingUnit = subset.Where(cm => cm.MappedColumn.Unit.SelectedUnit == UnitDescription.InvalidUnit && (cm.MappedColumn.ErrorStdDev == null || cm.MappedColumn.ErrorStdDev.Equals(Constants.STD_DEV_ARITHMETIC))).Select(cm => cm.MappedColumn.Name)
+            //all the mappings where the unit is undefined and 1. the error is not geometric or 2. the mapped column is empty
+            MissingUnit = subset
```

Comment:
Aside from the  code hard to understand, I don't understand why this fixes the issue.
You are checking now for the case where the unit was mapped to another column right? But in the issue, this was dealing with unit being not selected. How does it relate? The first condition should still have been Invalid no?

Reply:
No the Selected unit is set to invalid by creation. The problem is that also "get unit by column" was selected, but no column was mapped (ColumnName = ""). This is why this solves the problem